screen-weather-get.py

```diff
@@ -1,200 +1,69 @@
+def get_cached_weather(filename, ttl):
+    if is_stale(filename, ttl):
+        logging.info("Weather cache is stale.")
+        return None
 
-    return weather
+    logging.info("Found in cache")
+    with open(filename, 'r') as file:
+        return json.loads(file.read())
+    
 
-def get_response_data(url, filepath, ttl):
+def cache_weather_data(filename, weather_data):
+    if weather_data:
+        with open(filename, 'w') as text_file:
+            json.dump(weather_data, text_file)
 
-    response_json = False
-
-    if (is_stale(filepath, ttl)):
-        try:
-            response_data = requests.get(url).text
-            response_json = json.loads(response_data)['data']
-            with open(filepath, 'w') as text_file:
-                text_file.write(response_data)
-        except Exception as error:
-            logging.error(error)
-            raise
-    else:
-        with open(filepath, 'r') as file:
-            return json.loads(file.read())['data']
-    return response_json
 
 def main():
 
     # gather relevant environment configs
-
-    climacell_apikey=os.getenv("CLIMACELL_APIKEY","")
-
-    if climacell_apikey=="":
-        logging.error("CLIMACELL_APIKEY is missing")
+    climacell_apikey=os.getenv("CLIMACELL_APIKEY")
+    openweathermap_apikey=os.getenv("OPENWEATHERMAP_APIKEY")
+    
+    if not climacell_apikey and not openweathermap_apikey:
+        logging.error("API Key for weather is missing (Climacell, OpenWeatherMap)")
         sys.exit(1)
 
     weather_format=os.getenv("WEATHER_FORMAT", "CELSIUS")
-
     if (weather_format == "CELSIUS"):
         units = "metric"
     else:
         units = "imperial"
 
-    template_svg_filename = 'screen-template.svg'
-    output_svg_filename = 'screen-output-weather.svg'
-
-    # json response files
-    weather_timelines_filename = 'climacell-timelines-response.json'
-
-    location_lat = os.getenv("WEATHER_LATITUDE","51.3656") 
-    location_long = os.getenv("WEATHER_LONGITUDE","-0.1963") 
-
-    location_latlong = (
-        "{0:.2f},{1:.2f}"
-        .format(float(location_lat), float(location_long)))
+    location_lat = os.getenv("WEATHER_LATITUDE","51.3656")
+    location_long = os.getenv("WEATHER_LONGITUDE","-0.1963")
 
     # TTL for refetching of JSON
     ttl = float(os.getenv("WEATHER_TTL", 1 * 60 * 60))
+    cache_weather_file = "weather-cache.json"
+
+    if openweathermap_apikey:
+        logging.info("Getting weather from OpenWeatherMap")
+        weather = openweathermap.get_weather(openweathermap_apikey, location_lat, location_long, units)
+        logging.debug(weather)
+    elif climacell_apikey:
+        logging.info("Getting weather from Climacell")
 
-    logging.info("Gathering weather")
+        weather = get_cached_weather(cache_weather_file, ttl)
+        if not weather:
+            weather = climacell.get_weather(climacell_apikey, location_lat, location_long, units)
+            cache_weather_data(cache_weather_file, weather)
```

Comment:
Moved the cache check and cache storage to this 'calling' code, so the weather providers only need to make their API call, transform the data, and return it back.